CMake/RealmCore.cmake

```diff
@@ -237,9 +237,9 @@ function(build_realm_sync sync_directory)
     set_property(TARGET realm-sync-server PROPERTY IMPORTED_LOCATION_RELEASE ${sync_server_library_release})
     set_property(TARGET realm-sync-server PROPERTY IMPORTED_LOCATION ${sync_server_library_release})
     if(APPLE)
-        set_property(TARGET realm-sync-server PROPERTY INTERFACE_LINK_LIBRARIES ${FOUNDATION} ${SECURITY})
+        set_property(TARGET realm-sync-server PROPERTY INTERFACE_LINK_LIBRARIES ${FOUNDATION} ${SECURITY} -lyaml-cpp)
```

Comment:
In the meantime this can check if yaml-cpp is installed and only try to link it if so with `find_package(PkgConfig); pkg_check_modules(YAML QUIET yaml-cpp); set_property(TARGET realm-sync-server PROPERTY INTERFACE_LINK_LIBRARIES ${FOUNDATION} ${SECURITY} ${YAML_LDFLAGS})`.